src/DesignTime/QuotationsFactory.fs

```diff
@@ -660,24 +660,14 @@ type internal QuotationsFactory private() =
                 returnTypes
                 |> List.mapi (fun i rt ->
                     let propName = sprintf "ResultSet%d" (i + 1)
-                    let prop = ProvidedProperty(propName, rt.Single, fun args -> <@@ %%args.[0] |> unbox |> Map.find<string, obj> propName @@>)
+                    let prop = ProvidedProperty(propName, rt.Single, fun args -> <@@ (%%args.[0] |> unbox<obj[]>).[i] @@>)
                     let ctorParam = ProvidedParameter(propName, rt.Single)
                     prop, ctorParam)
                 |> List.unzip
 
             resultSetsType.AddMembers props
 
-            let invokeCode args =
```

Comment:
This technique should be usable for the record result type too. No need to bake in property names, so quotations passed around will be much smaller. At runtime, property access and "record" instantiation will be dozens of times faster while consuming a lot less memory. It should be really noticeable when querying lots of records.